src/components/Chat.tsx

```diff
@@ -92,17 +93,34 @@ const Chat: React.FC<ChatProps> = ({
     };
   }, [eventTarget]);
 
+  // Sync selected text contexts from plugin
+  useEffect(() => {
+    const updateSelectedTextContexts = () => {
+      setSelectedTextContexts([...plugin.selectedTextContexts]);
+    };
+
+    // Initialize with current state
+    updateSelectedTextContexts();
+
+    // Subscribe to changes
+    const unsubscribe = plugin.subscribeToSelectedTextContextsChange(updateSelectedTextContexts);
```

Comment:
We can create an atom to communicate the selected text with the component. I find the subscription model overkill and tedious to maintain. With atom, you can have the command set the value, and have this component use the atom value.